contracts/ReputationMiningCycleRespond.sol

```diff
@@ -107,6 +123,29 @@ contract ReputationMiningCycleRespond is ReputationMiningCycleStorage, PatriciaT
     u[U_DECAY_TRANSITION] = 0;
     u[U_GLOBAL_CHILD_UPDATE] = 0;
     u[U_NEW_REPUTATION] = 0;
+    // For functions that want the keys directly, we build them here (once) and then can pass them the array.
+    bytes[5] memory keys;
+    keys[0] = buildReputationKey(b32[B_REPUTATION_KEY_COLONY], b32[B_REPUTATION_KEY_SKILLID], b32[B_REPUTATION_KEY_USER]);
+    keys[1] = buildReputationKey(
+      b32[B_PREVIOUS_NEW_REPUTATION_KEY_COLONY],
+      b32[B_PREVIOUS_NEW_REPUTATION_KEY_SKILLID],
+      b32[B_PREVIOUS_NEW_REPUTATION_KEY_USER]
+    );
+    keys[2] = buildReputationKey(
+      b32[B_ADJACENT_REPUTATION_KEY_COLONY], 
+      b32[B_ADJACENT_REPUTATION_KEY_SKILLID], 
+      b32[B_ADJACENT_REPUTATION_KEY_USER]
+    );
+    keys[3] = buildReputationKey(
+      b32[B_ORIGIN_ADJACENT_REPUTATION_KEY_COLONY],
+      b32[B_ORIGIN_ADJACENT_REPUTATION_KEY_SKILLID],
+      b32[B_ORIGIN_ADJACENT_REPUTATION_KEY_USER]
+    );
+    keys[4] = buildReputationKey(
+      b32[B_CHILD_ADJACENT_KEY_COLONY],
+      b32[B_CHILD_ADJACENT_KEY_SKILLID],
+      b32[B_CHILD_ADJACENT_KEY_USER]
+    );
```

Comment:
After staring at this for a while, I think you're right. Fundamentally, we can't check anything about these keys (other than that they exist in the relevant trees, which we do via `getImpliedRoot` after hashing).

Reply:
The only exception is `checkKeyLogEntry`..

Reply:
Yeah, indeed. So I've changed this array to taking the three elements of `reputationKey`, and then the hashes of all five keys in question, one of which is `reputationKey` (because we use its hash a lot). I check the supplied hash is correct once, and then just use the supplied hash elsewhere directly.